flake.lock

```diff
@@ -359,16 +360,16 @@
     "hls-2.4": {
```

Comment:
I have no idea what I'm talking about, but just in case: should the version here match the update in `shell.nix`?

Reply:
This file is the [Nix flake lock file](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-flake#lock-files), which contains the exact pins (eg git commit hashes + further integrity hashes) of everything external used by our flake here, and also all inputs used transitively by flakes we use etc. In this case, `hls-2.4` is *not* a direct input of our flake, but rather a transitive input via [haskell.nix](https://github.com/input-output-hk/haskell.nix).[^1]

We don't use this input ourselves currently (we use HLS 2.5 from Hackage)[^2], so there is no need to adapt anything to stay in sync with our declared version 2.5.0.0 in `shell.nix`. The flake lock file is usually updated automatically by commands like `nix flake update` which I ran for this PR, which updated haskell.nix and hence also this transitive flake input, which is why it shows up in the diff.

Hope that makes a bit of sense :smiley: 

[^1]: You can explore the flake graph interactively with eg [nix-melt](https://github.com/nix-community/nix-melt).
[^2]: Motivation why haskell.nix has this input: In the past, HLS didn't compile if you didn't use its own [`cabal.project`](https://github.com/haskell/haskell-language-server/blob/master/cabal.project) (which you don't get if you compile it from Hackage as we do ATM), so haskell.nix pinned HLS from GitHub which contains the `cabal.project`.

Reply:
Thank you for explaining!